src/TestFramework/TestFramework/Assertions/Assert.IsInstanceOfType.cs

```diff
@@ -260,9 +258,8 @@ public static void IsNotInstanceOfType<T>(object? value, string? message)
             return;
         }
 
-        TypeInfo elementTypeInfo = value.GetType().GetTypeInfo();
-        TypeInfo expectedTypeInfo = wrongType.GetTypeInfo();
-        if (expectedTypeInfo.IsAssignableFrom(elementTypeInfo))
+        Type elementType = value.GetType();
+        if (elementType.IsAssignableFrom(wrongType))
```

Comment:
This is backwards, no?
```suggestion
        if (wrongType.IsAssignableFrom(elementType))
```